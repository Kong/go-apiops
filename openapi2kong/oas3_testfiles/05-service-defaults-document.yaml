# x-kong-service-defaults can be specified and will be honored on each level

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/
x-kong-service-defaults:
  retries: 100
paths:
  /path1:
    # this should create routes attached to the document-level service
    get:
      operationId: uses-doc-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    # specify server block to force another service being created
    # should inherit the service defaults from the document level
    servers:
      - url: https://server2.com/
    get:
      operationId: uses-path-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
    post:
      # specify server block to force another service being created
      # should inherit the service defaults from the document level
      servers:
        - url: https://server3.com/
      operationId: uses-ops-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
