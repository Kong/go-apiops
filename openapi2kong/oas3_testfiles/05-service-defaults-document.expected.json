{
  "_format_version": "3.0",
  "services": [
    {
      "host": "server1.com",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "retries": 100,
      "routes": [
        {
          "id": "663104d8-7e60-525d-b506-e42971b4466b",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-doc-service",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_05-service-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_05-service-defaults-document.yaml"
      ]
    },
    {
      "host": "server2.com",
      "id": "a79c5a8c-0924-599e-9412-39f5a4ff0c3e",
      "name": "simple-api-overview_path2",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "retries": 100,
      "routes": [
        {
          "id": "c1b84366-8ff8-57b9-b118-bedd1b9ab1c8",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-path-service",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_05-service-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_05-service-defaults-document.yaml"
      ]
    },
    {
      "host": "server3.com",
      "id": "5d05e040-c551-50a1-90e7-6af5d3a7c0dc",
      "name": "simple-api-overview_uses-ops-service",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "retries": 100,
      "routes": [
        {
          "id": "ca1c6faa-7076-5a09-9346-9c9dc3e019cb",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_uses-ops-service",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_05-service-defaults-document.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_05-service-defaults-document.yaml"
      ]
    }
  ],
  "upstreams": []
}